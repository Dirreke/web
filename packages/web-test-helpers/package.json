{
  "name": "web-test-helpers",
  "private": true,
  "peerDependencies": {
      "@vue/composition-api": "1.7.1",
      "vue-gettext": "^2.1.5",
      "vue-router": "3.6.5",
      "vuex": "^3.6.2"
  }
}
