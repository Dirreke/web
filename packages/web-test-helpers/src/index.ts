export * from './mocks/defaultComponentMocks'
export * from './mocks/defaultStubs'
export * from './mocks/axios'

export * from './defaultPlugins'
export * from './helpers'
