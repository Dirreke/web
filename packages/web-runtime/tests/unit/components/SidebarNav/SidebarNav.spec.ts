import SidebarNav from 'web-runtime/src/components/SidebarNav/SidebarNav.vue'
import stubs from '../../../../../../tests/unit/stubs'
import sidebarNavItemFixtures from '../../../__fixtures__/sidebarNavItems'
import { createStore, defaultPlugins, mount } from 'web-test-helpers'

jest.mock('uuid', () => ({
  v4: () => {
    return '00000000-0000-0000-0000-000000000000'
  }
}))

const slots = {
  bottom: '<span class="footer">Footer</span>'
}

describe('OcSidebarNav', () => {
  const { wrapper } = getWrapper()

  it('displays a bottom slot if given', () => {
    expect(wrapper.findAll('.footer').length).toBe(1)
  })

  it('renders navItems into a list', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('toggles into closed state upon button click', async () => {
    await wrapper.find('.toggle-sidebar-button').trigger('click')
    await wrapper.vm.$nextTick()
    expect(wrapper).toMatchSnapshot()
  })

  it('toggles back into open state upon button click', async () => {
    await wrapper.find('.toggle-sidebar-button').trigger('click')
    await wrapper.vm.$nextTick()
    expect(wrapper).toMatchSnapshot()
  })
})

function getWrapper() {
  const storeOptions = {
    state: {
      navigation: {
        closed: false
      }
    },
    mutations: {
      SET_CLOSED(state, value) {
        state.navigation.closed = value
      }
    },
    actions: {
      openNavigation({ commit }) {
        commit('SET_CLOSED', false)
      },
      closeNavigation({ commit }) {
        commit('SET_CLOSED', true)
      }
    }
  }
  const store = createStore(storeOptions)
  return {
    wrapper: mount(SidebarNav, {
      slots,
      props: {
        navItems: sidebarNavItemFixtures
      },
      global: {
        plugins: [...defaultPlugins(), store],
        stubs: {
          ...stubs,
          'sidebar-nav-item': true,
          'oc-button': false
        }
      }
    })
  }
}
