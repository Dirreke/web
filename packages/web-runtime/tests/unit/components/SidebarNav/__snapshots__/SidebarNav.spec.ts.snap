// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OcSidebarNav renders navItems into a list 1`] = `
<div class="oc-app-navigation-expanded" id="web-nav-sidebar">
  <button aria-label="Toggle sidebar" class="toggle-sidebar-button oc-pb-s oc-pt-m oc-button oc-rounded oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw toggle-sidebar-button-expanded oc-pr-s" type="button">
    <oc-icon-stub accessiblelabel="" color="" filltype="line" name="arrow-drop-left" size="large" type="span" variation="passive"></oc-icon-stub>
  </button>
  <nav aria-label="Sidebar navigation menu" class="oc-sidebar-nav oc-mb-m oc-mt-s oc-px-xs">
    <oc-list-stub>
      <div class="oc-ml-s oc-background-primary-gradient" id="nav-highlighter"></div>
      <sidebar-nav-item-stub active="true" filltype="fill" icon="folder" index="00000000000000000000000000000000" name="Personal" target="/files/list/all"></sidebar-nav-item-stub>
      <sidebar-nav-item-stub filltype="fill" icon="share-forward" index="00000000000000000000000000000000" name="Shares" target="/files/list/shared-with-me"></sidebar-nav-item-stub>
      <sidebar-nav-item-stub filltype="fill" icon="delete" index="00000000000000000000000000000000" name="Deleted files" target="/files/list/trash-bin"></sidebar-nav-item-stub>
    </oc-list-stub>
  </nav>
  <span class="footer">Footer</span>
</div>
`;

exports[`OcSidebarNav toggles back into open state upon button click 1`] = `
<div class="oc-app-navigation-expanded" id="web-nav-sidebar">
  <button aria-label="Toggle sidebar" class="toggle-sidebar-button oc-pb-s oc-pt-m oc-button oc-rounded oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw toggle-sidebar-button-expanded oc-pr-s" type="button">
    <oc-icon-stub accessiblelabel="" color="" filltype="line" name="arrow-drop-left" size="large" type="span" variation="passive"></oc-icon-stub>
  </button>
  <nav aria-label="Sidebar navigation menu" class="oc-sidebar-nav oc-mb-m oc-mt-s oc-px-xs">
    <oc-list-stub>
      <div class="oc-ml-s oc-background-primary-gradient" id="nav-highlighter"></div>
      <sidebar-nav-item-stub active="true" filltype="fill" icon="folder" index="00000000000000000000000000000000" name="Personal" target="/files/list/all"></sidebar-nav-item-stub>
      <sidebar-nav-item-stub filltype="fill" icon="share-forward" index="00000000000000000000000000000000" name="Shares" target="/files/list/shared-with-me"></sidebar-nav-item-stub>
      <sidebar-nav-item-stub filltype="fill" icon="delete" index="00000000000000000000000000000000" name="Deleted files" target="/files/list/trash-bin"></sidebar-nav-item-stub>
    </oc-list-stub>
  </nav>
  <span class="footer">Footer</span>
</div>
`;

exports[`OcSidebarNav toggles into closed state upon button click 1`] = `
<div class="oc-app-navigation-collapsed" id="web-nav-sidebar">
  <button aria-label="Toggle sidebar" class="toggle-sidebar-button oc-pb-s oc-pt-m oc-button oc-rounded oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw toggle-sidebar-button-collapsed" type="button">
    <oc-icon-stub accessiblelabel="" color="" filltype="line" name="arrow-drop-right" size="large" type="span" variation="passive"></oc-icon-stub>
  </button>
  <nav aria-label="Sidebar navigation menu" class="oc-sidebar-nav oc-mb-m oc-mt-s oc-px-xs">
    <oc-list-stub>
      <div class="oc-ml-s oc-background-primary-gradient" id="nav-highlighter"></div>
      <sidebar-nav-item-stub active="true" collapsed="true" filltype="fill" icon="folder" index="00000000000000000000000000000000" name="Personal" target="/files/list/all"></sidebar-nav-item-stub>
      <sidebar-nav-item-stub collapsed="true" filltype="fill" icon="share-forward" index="00000000000000000000000000000000" name="Shares" target="/files/list/shared-with-me"></sidebar-nav-item-stub>
      <sidebar-nav-item-stub collapsed="true" filltype="fill" icon="delete" index="00000000000000000000000000000000" name="Deleted files" target="/files/list/trash-bin"></sidebar-nav-item-stub>
    </oc-list-stub>
  </nav>
  <span class="footer">Footer</span>
</div>
`;
